xonsh/environ.py

```diff
@@ -691,18 +691,36 @@ def locate_binary(name):
         return None
 
 
+def _get_parent_dir_for(cwd, dir_name):
+    # walk up the directory tree to see if we are inside an hg repo
+    dir_list = [dir_name]
+    path = cwd.split(os.path.sep)
```

Comment:
>You should use os.path.split and os.path.join

I thought about using `os.path.join` and because I had to spread the arguments, I decided not to change this part. I'll do it though. For `os.path.split`, I don't think it works as you expect it to work. It just splits the path into two.